### PR TITLE
Implement search users feature

### LINKED ISSUE #344: Implement search option in users tab
**Actual Behaviour**
There is no option to search users on users tab

**Expected Behaviour**
There should be a way to search for users

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by PrP-11**
Fixes #344 

Changes: Implement search feature on users tab

Surge Deployment Link: https://pr-381-fossasia-susi-accounts.surge.sh

Screenshots for the change:
s1

### CODE REVIEW

File: src/components/Admin/ListUser/ListUser.js [hunk 1]
```diff
@@ -21,6 +22,8 @@ const cookies = new Cookies();
 
 const TabPane = Tabs.TabPane;
 
+const Search = Input.Search;
```
Comment (anshumanv) [inline src/components/Admin/ListUser/ListUser.js]: `const Search = { Input }`

File: src/components/Admin/ListUser/ListUser.js [hunk 2]
```diff
@@ -184,6 +188,56 @@ export default class ListUser extends Component {
     });
   };
 
+  handleSearch = value => {
+    this.setState({
+      search: true,
+    });
+    let url;
```
Comment (anshumanv) [inline src/components/Admin/ListUser/ListUser.js]:
```
let url = `...`
```

If you're using template literals then you won't need `+` to combine urls/params

File: src/components/Admin/ListUser/ListUser.js [hunk 3]
```diff
@@ -184,6 +188,56 @@ export default class ListUser extends Component {
...
+      jsonp: 'callback',
+      crossDomain: true,
+      success: function(response) {
+        let userList = response.users;
+        let users = [];
+        userList.map((data, i) => {
+          let user = {
+            serialNum: ++i,
+            email: data.name,
+            confirmed: data.confirmed,
```
Comment (anshumanv) [inline src/components/Admin/ListUser/ListUser.js]: No point of storing this if you're changing it later anyways.

File: src/components/Admin/ListUser/ListUser.js [hunk 4]
```diff
@@ -184,6 +188,56 @@ export default class ListUser extends Component {
...
+            signup: data.signupTime,
+            lastLogin: data.lastLoginTime,
+            ipLastLogin: data.lastLoginIP,
+            userRole: data.userRole,
+          };
+
+          if (user.confirmed) {
```
Comment (anshumanv) [inline src/components/Admin/ListUser/ListUser.js]: `data.confirmed`

File: src/components/Admin/ListUser/ListUser.js [hunk 5]
```diff
@@ -474,16 +528,31 @@ export default class ListUser extends Component {
                           </div>
                         </Dialog>
                       </div>
-
+                      <Search
+                        placeholder="search by email"
```
Comment (anshumanv) [inline src/components/Admin/ListUser/ListUser.js]: `Seach by email`

File: src/components/Admin/ListUser/ListUser.js [hunk 6]
```diff
@@ -474,16 +528,31 @@ export default class ListUser extends Component {
...
+                        style={{ margin: '5px 25% 20px 25%', width: '50%' }}
+                        size="default"
+                        onSearch={value => this.handleSearch(value)}
```
Comment (anshumanv) [inline src/components/Admin/ListUser/ListUser.js]: Is there some event like `onChange` which filters the user as the search field is changed? This looks good too though.

File: src/components/Admin/ListUser/ListUser.js (hunk 2 above)
Comment (PrP-11) [reply src/components/Admin/ListUser/ListUser.js]: Fixed it.

File: src/components/Admin/ListUser/ListUser.js (hunk 3 above)
Comment (PrP-11) [reply src/components/Admin/ListUser/ListUser.js]: Removed it

File: src/components/Admin/ListUser/ListUser.js (hunk 6 above)
Comment (PrP-11) [reply src/components/Admin/ListUser/ListUser.js]: @anshumanv The data is not filtered on the front end but instead an api is called to return the desired search results so its better if we call api only on click or enter

File: src/components/Admin/ListUser/ListUser.js (hunk 1 above)
Comment (PrP-11) [reply src/components/Admin/ListUser/ListUser.js]: @anshumanv antdesign follows the above syntax and changing it to `const Search = { Input }` will throw errors on the console

File: src/components/Admin/ListUser/ListUser.js (hunk 6 above)
Comment (anshumanv) [reply src/components/Admin/ListUser/ListUser.js]: Okay then, works fine either way. :+1:

File: src/components/Admin/ListUser/ListUser.js (hunk 1 above)
Comment (anshumanv) [reply src/components/Admin/ListUser/ListUser.js]: Okay, I see. Looks good then. :+1:

Comment (anshumanv) [reply src/components/Admin/ListUser/ListUser.js]: Can you please post the error? I'm interested in knowing what it throws. @PrP-11

Comment (anshumanv) [reply src/components/Admin/ListUser/ListUser.js]: Thanks @PrP-11  :sparkle:

**Pull request merged**